model/prelude.sail

```diff
@@ -191,114 +248,112 @@ val EXTZ : forall 'n 'm, 'm >= 'n. (implicit('m), bits('n)) -> bits('m)
 function EXTS(m, v) = sail_sign_extend(v, m)
 function EXTZ(m, v) = sail_zero_extend(v, m)
 
-val zeros_implicit : forall 'n, 'n >= 0 . implicit('n) -> bits('n)
-function zeros_implicit (n) = sail_zeros(n)
+val zeros_implicit : forall 'n, 'n >= 0. implicit('n) -> bits('n)
+function zeros_implicit n = sail_zeros(n)
 overload zeros = {zeros_implicit}
 
-val ones : forall 'n, 'n >= 0 . implicit('n) -> bits('n)
-function ones (n) = sail_ones (n)
+val ones : forall 'n, 'n >= 0. implicit('n) -> bits('n)
+function ones n = sail_ones(n)
 
 val bool_to_bits : bool -> bits(1)
-function bool_to_bits x = if x then 0b1 else 0b0
+function bool_to_bits x = if x then {
+  0b1
+} else {
+  0b0
+}
 
 val bit_to_bool : bit -> bool
 function bit_to_bool b = match b {
-  bitone  => true,
-  bitzero => false
+  bitone => true,
+  bitzero => false,
 }
 
-val to_bits : forall 'l, 'l >= 0.(atom('l), int) -> bits('l)
-function to_bits (l, n) = get_slice_int(l, n, 0)
+val to_bits : forall 'l, 'l >= 0. (int('l), int) -> bits('l)
+function to_bits(l, n) = get_slice_int(l, n, 0)
```

Comment:
I'm not sure why you would have a space though? Does anyone format Rust/C/Java like:
```
int f (int x) {

}
```
I can add an option to force a space between a function name and it's parameter list but it seems to me like an odd formatting choice to make function declaration `function f (x, y)` and function calling `f(x, y)` inconsistent. If I was being opinionated I would force single argument functions to be `f(x)` rather than allowing the `f x` sugar for a single argument, because while omitting the brackets in something like `function clause decode 0b00000 @ ... @ ... = <exp>` is quite nice, I still think that sugar is a bit too sweet for my tastes, and in retrospect I think I should have forced the parser to be more strict.

Reply:
I have a slight preference towards always using parens without a space (even with a single argument) since it matches other languages.

Reply:
Well, because it's not the same; this is like functional languages where multiple arguments are done by virtue of having pattern matching and tuples. Grammatically/syntactically you're passing a single tuple argument that is being pattern matched and deconstructed, so it's no different from `function f PAIR(x, y)` which _doesn't_ get parentheses around it. In other languages you use `f(x, y)` and `f(x)` because `f(...)` is the syntax for function arguments.

Reply:
Semantically Sail functions do have multiple arguments, so if you want to pass a tuple to a function you do have to do `f((x, y))`, for example:
```
$include <prelude.sail>

val fst : forall ('a 'b: Type). (('a, 'b)) -> 'a

// The `function f x` sugar lets us treat (x, y) as a single pattern
function fst(x, y) = x

val main : unit -> unit

function main() = {
  print_int("", fst((1, 3)))
}
```
Things are set up this way so we can translate a Sail function `(a, b) -> c` into an ML-ish target as either uncurried `a * b -> c` or curried `a -> b -> c` depending on what is more idiomatic.

Reply:
Huh, interesting, I always assumed it was ML-like. Fair enough then.

Reply:
In the very very distant past (like 4-5ish years ago I think) it was, but it made generating idiomatic theorem prover code that used curried functions more difficult. Tagged union constructors are still unary (like in OCaml) and they have syntax sugar, so:
```
$include <prelude.sail>

union U('a: Type, 'b: Type) = {
    Pair : ('a, 'b)
}

val main : unit -> unit

function main() = {
    let p: (int, int) = (1, 2);
    let _ = Pair(p);
    let _: U(int, int) = Pair(1, 2); // syntax sugar to avoid nested brackets
    // equivalent to
    let _: U(int, int) = Pair((1, 2));
}
```
I've now written a lot of Rust, and it does force you to write `Constructor((x, y))` and I haven't found it annoying at all, so that's made me reconsider a bit whether this type of syntax sugar was really a good idea in Sail.

Reply:
I generally like my code formatters to have options though, so I'll probably add a toggle option for this type of syntax sugar.